### PR TITLE
delete invitations to a non-existent VC / when deleting VC

### LINKED ISSUE #4308: BUG: When a VC has open invitation to a community and the VC is removed, the Community page to which has invitation fails to load
**Describe the bug**

**To Reproduce**
1. Go to any env
2. Invite a VC from a space
3. Delete the VC
4. Navigate to the space /settings/community - applications/invitations data fails to be retrieved

**Expected behavior**
Applications/invitations data is retrieved without errors.

### PR DESCRIPTION
**Pull request by valentinyanakiev**
- uses entity manager in VC service due to spaghettin dependencies. Hell, I tried to fix those, but they ran deeper than I expected
- removes all phantom invitations to VCs on the platform via migration
- API test already merged to `develop` on `test-suites` - `npm run test:usermanagement ./test/functional-api/user-management/virtual-contributor/`

resolves #4308

### CODE REVIEW

File: src/domain/community/virtual-contributor/virtual.contributor.service.ts [hunk 1]
```diff
@@ -546,4 +551,17 @@ export class VirtualContributorService {
 
     return virtualContributorMatchesCount;
   }
+
+  //adding this to avoid circular dependency between VirtualContributor, Room, and Invitation
+  private async deleteVCInvitations(contributorID: string) {
+    const invitations = await this.entityManager.find(Invitation, {
+      relations: { community: true },
```
Comment (hero101) [inline src/domain/community/virtual-contributor/virtual.contributor.service.ts]: why do you need that loaded?

File: src/domain/community/virtual-contributor/virtual.contributor.service.ts [hunk 2]
```diff
@@ -546,4 +551,17 @@ export class VirtualContributorService {
...
+      where: { invitedContributor: contributorID },
+    });
+    for (const invitation of invitations) {
+      if (invitation.authorization)
+        await this.authorizationPolicyService.delete(invitation.authorization);
+      await this.entityManager.remove<Invitation>(invitation);
```
Comment (hero101) [inline src/domain/community/virtual-contributor/virtual.contributor.service.ts]: i don't think you need `Invitation` here, furthermore the type should be infered by the type of `invitation`

File: src/domain/community/virtual-contributor/virtual.contributor.service.ts [hunk 3]
```diff
@@ -546,4 +551,17 @@ export class VirtualContributorService {
...
   }
+
+  //adding this to avoid circular dependency between VirtualContributor, Room, and Invitation
+  private async deleteVCInvitations(contributorID: string) {
+    const invitations = await this.entityManager.find(Invitation, {
+      relations: { community: true },
+      where: { invitedContributor: contributorID },
+    });
+    for (const invitation of invitations) {
+      if (invitation.authorization)
```
Comment (hero101) [inline src/domain/community/virtual-contributor/virtual.contributor.service.ts]: braces, please

**Review by hero101 — CHANGES_REQUESTED** [review]

File: src/domain/community/virtual-contributor/virtual.contributor.service.ts (hunk 1 above)
Comment (valentinyanakiev) [reply src/domain/community/virtual-contributor/virtual.contributor.service.ts]: I do not, will remove it.

**Pull request merged**